tests/e2e/system_test/test_full_cluster_health.py

```diff
@@ -0,0 +1,306 @@
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, request, nodes):
+        """
+        Setting up test parameters
+        """
+
+        def teardown():
+            logger.info("cleanup the environment")
+            self.operator.cleanup()
+            logger.info(
+                "Going to sleep for 5 Minute, for background cleanup to complete"
+            )
+            time.sleep(300)
+            nodes.restart_nodes_by_stop_and_start_teardown()
+
+        request.addfinalizer(teardown)
+
+        logger.info("Starting the test setup")
+        self.percent_to_fill = 85.0
+        self.ceph_cluster = CephCluster()
+        self.nodes = None
+
+        self.benchmark_name = "FIO"
+        self.client_pod_name = "fio-client"
+
+        self.sanity_helpers = sanity_helpers.Sanity()
+
+        super(TestFullClusterHealth, self).setup()
+        # deploy the benchmark-operator
+        self.deploy_benchmark_operator()
+
+    def run(self):
+        """
+
+        Run the test, and wait until it finished
+        """
+
+        self.deploy_and_wait_for_wl_to_start(timeout=900)
+        self.wait_for_wl_to_finish(sleep=300)
+
+        try:
+            if "Fio failed to execute" not in self.test_logs:
+                logger.info("FIO has completed successfully")
+        except IOError:
+            logger.warning("FIO failed to complete")
+
+    def calculate_crd_data(self):
+        """
+        Getting the storage capacity and calculate pod count and pvc size
+
+        """
+
+        ceph_used_capacity_percent = get_percent_used_capacity()
+        logging.info(f"Ceph used capacity percent is {ceph_used_capacity_percent}%")
+
+        ceph_capacity = self.ceph_cluster.get_ceph_capacity()
+        logging.info(f"Total storage capacity is {ceph_capacity} GiB")
+
+        self.percent_to_fill = self.percent_to_fill - ceph_used_capacity_percent
+        logging.info(f"Percentage to fill is {self.percent_to_fill}%")
+
+        self.total_data_set = int(ceph_capacity * (int(self.percent_to_fill) / 100))
+        self.filesize = int(
+            self.crd_data["spec"]["workload"]["args"]["filesize"].replace("GiB", "")
+        )
+
+        # Make sure that filesize>=10 and servers<=60
+        self.servers = 60
+        self.filesize = int(self.total_data_set / self.servers)
+        if self.filesize < 10:
+            self.filesize = 10
+            self.servers = int(self.total_data_set / self.filesize)
+
+        self.crd_data["spec"]["workload"]["args"]["filesize"] = f"{self.filesize}GiB"
+        self.crd_data["spec"]["workload"]["args"][
+            "storagesize"
+        ] = f"{int(self.total_data_set)}Gi"
+        self.crd_data["spec"]["workload"]["args"]["servers"] = self.servers
+        self.crd_data["spec"]["workload"]["args"]["bs"] = "1024KiB"
+        self.crd_data["spec"]["workload"]["args"]["jobs"] = ["write", "read"]
+        self.crd_data["spec"]["workload"]["args"]["iodepth"] = 1
+
+    def cleanup(self):
```

Comment:
If all these resources are created in the benchmark_operator namespace you can actually remove the entire clean up function. benchmark_operator cleanup() in teardown will take care of deleting the created resources.

Reply:
Removed